fenix/app/src/main/java/org/mozilla/fenix/translations/preferences/downloadlanguages/DownloadLanguagesPreference.kt

```diff
@@ -0,0 +1,508 @@
+    }
+
+    val allLanguagesItem = downloadLanguageItemPreferences.last {
+        it.state.type == DownloadLanguageItemTypePreference.AllLanguages
+    }
+
+    val defaultLanguageItem = downloadLanguageItemPreferences.last {
+        it.state.type == DownloadLanguageItemTypePreference.PivotLanguage
+    }
+
+    Column(
+        modifier = Modifier.background(
+            color = FirefoxTheme.colors.layer1,
+        ),
+    ) {
+        DownloadLanguagesHeaderPreference()
+
+        LazyColumn {
+            item {
+                AvailableLanguagesHeader()
+            }
+
+            item {
+                LanguageItemPreference(
+                    item = defaultLanguageItem,
+                    onItemClick = onItemClick,
+                )
+            }
+
+            items(downloadedItems) { item: DownloadLanguageItemPreference ->
+                LanguageItemPreference(
+                    item = item,
+                    onItemClick = onItemClick,
+                )
+            }
+
+            items(inProgressItems) { item: DownloadLanguageItemPreference ->
+                LanguageItemPreference(
+                    item = item,
+                    onItemClick = onItemClick,
+                )
+            }
+
+            item {
+                Divider(Modifier.padding(top = 8.dp, bottom = 8.dp))
+            }
+
+            item {
+                DownloadLanguagesHeader()
+            }
+
+            item {
+                LanguageItemPreference(
+                    item = allLanguagesItem,
+                    onItemClick = onItemClick,
+                )
+                Divider(Modifier.padding(top = 8.dp, bottom = 8.dp))
+            }
+
+            items(notDownloadedItems) { item: DownloadLanguageItemPreference ->
+                item.languageModel.language?.localizedDisplayName?.let {
+                    LanguageItemPreference(
+                        item = item,
+                        onItemClick = onItemClick,
+                    )
+                }
+            }
+        }
+    }
+}
+
+@Composable
+private fun DownloadLanguagesHeader() {
+    Text(
+        text = stringResource(
+            id = R.string.download_language_header_preference,
+        ),
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(start = 72.dp, end = 16.dp, top = 8.dp)
+            .semantics { heading() },
+        color = FirefoxTheme.colors.textAccent,
+        style = FirefoxTheme.typography.headline8,
+    )
+}
+
+@Composable
+private fun AvailableLanguagesHeader() {
+    Text(
+        text = stringResource(
+            id = R.string.download_languages_available_languages_preference,
+        ),
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(start = 72.dp, end = 16.dp, top = 8.dp)
+            .semantics { heading() },
+        color = FirefoxTheme.colors.textAccent,
+        style = FirefoxTheme.typography.headline8,
+    )
+}
```

Comment:
Can we add a reusable header component since these are styled the same?

Reply:
Done